sky/serve/load_balancer.py

```diff
@@ -160,11 +160,28 @@ async def _proxy_with_retries(
         # SkyServe supports serving on Spot Instances. To avoid preemptions
         # during request handling, we add a retry here.
         retry_cnt = 0
+        # Here we try to not retry those failed replicas for the case when the
+        # replica is in a NOT_READY state but does not sync-ed to the load
+        # balancer yet. However, we still maintain a per-request failed replica
+        # list instead of the global one to avoid the case for transient
+        # networking issues, and letting new requests to retry them. Since our
+        # LB policy is a global one, when the request rate is high, it is likely
+        # that multiple retries on a single request will use the same replica.
+        # Here we use the failed replica list to keep track of the failures
+        # happened on every request and try to avoid them in the next retry.
+        failed_replica_urls: List[str] = []
         while True:
             retry_cnt += 1
             with self._client_pool_lock:
                 ready_replica_url = self._load_balancing_policy.select_replica(
-                    request)
+                    request, failed_replica_urls)
+                # If all replicas are failed, retry them again as some
+                # of them might be transient networking issues.
+                if ready_replica_url is None and failed_replica_urls:
+                    failed_replica_urls = []
+                    ready_replica_url = (
+                        self._load_balancing_policy.select_replica(
+                            request, failed_replica_urls))
```

Comment:
This may increase the index twice for round robin. Should we instead, just check if `len(failed_replica_urls) == len(ready_replica)` (can add an interface for LBPolicy.num_ready_replica()` for latter) before we call the previous `select_replica` and set `failed_replica_urls` to empty before calling `select_replica`?

Reply:
Yeah that is a good call. Done!